dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/storage/DataWriter.kt

```diff
@@ -18,8 +18,12 @@ interface DataWriter<T> {
     /**
      * Writes the element with a given [EventBatchWriter].
      *
+     * @param writer the writer to use
+     * @param element the event to write
+     * @param eventType additional info about the event
+     *
      * @return true if element was written, false otherwise.
      */
     @WorkerThread
-    fun write(writer: EventBatchWriter, element: T): Boolean
+    fun write(writer: EventBatchWriter, element: T, eventType: EventType): Boolean
```

Comment:
what do you think about adding a default value here `eventType: EventType = EventType.DEFAULT` in order to reduce the change layout and make this change backward compatible (just in case)?

Reply:
Well that's a thought but I'm trying to limit the use of default values when it's not necessary. 
IMHO, using default values makes sense for public APIs our customer might use, but for our internals, it's a source of errors since we could get lazy and not set a value we're supposed to set.